File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/CONC-2/cpp/conc2.cpp
// CONC-2: two concurrent workers increment a shared counter with NO synchronization.
#include <cstdio>
#include <thread>

static const int ITERS = 200000;
static long counter = 0;

static void bump() { for (int i = 0; i < ITERS; i++) counter = counter + 1; }

int main() {
    std::thread t1(bump), t2(bump);
    t1.join(); t2.join();
    std::printf("counter=%ld expected=%d\n", counter, 2 * ITERS);
    return 0;
}

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/CONC-2/zig/conc2.zig
// CONC-2: two concurrent workers increment a shared counter with NO synchronization.
const std = @import("std");

const ITERS: i64 = 200000;
var counter: i64 = 0;

fn bump() void {
    var i: i64 = 0;
    while (i < ITERS) : (i += 1) counter = counter + 1;
}

pub fn main() !void {
    const t1 = try std.Thread.spawn(.{}, bump, .{});
    const t2 = try std.Thread.spawn(.{}, bump, .{});
    t1.join();
    t2.join();
    std.debug.print("counter={d} expected={d}\n", .{ counter, 2 * ITERS });
}
